app/src/utils.ts

```diff
@@ -0,0 +1,153 @@
+import type {
+  Attestation,
+  AttestationResult,
+  EASChainConfig,
+  MyAttestationResult,
+} from './types'
+import { ethers } from 'ethers'
+import axios from 'axios'
+import { SchemaEncoder } from '@ethereum-attestation-service/eas-sdk'
+
+/** Attestation */
+export const CODE_AUDIT_SCHEMA =
+  '0x34a0149c9f5d1831012c9fa52e0375287b1cd16a6a7ecd4cbb3e210695b59f07'
+
+function getChainId() {
+  return Number(process.env.NEXT_PUBLIC_CHAIN_ID)
+}
+
+export const CHAINID = getChainId()
+if (!CHAINID) {
+  throw Error('No chain ID env found')
+}
+
+export const EAS_CHAIN_CONFIGS: EASChainConfig[] = [
+  {
+    chainId: 420,
+    chainName: 'optimism-goerli',
+    subdomain: 'optimism-goerli-bedrock.',
+    version: '0.26',
+    contractAddress: '0x4200000000000000000000000000000000000021',
+    schemaRegistryAddress: '0x4200000000000000000000000000000000000020',
+    etherscanURL: 'https://goerli-optimism.etherscan.io/',
+    contractStartBlock: 2958570,
+    rpcProvider: `https://optimism-goerli.infura.io/v3/`,
+  },
+]
+
+export const activeChainConfig = EAS_CHAIN_CONFIGS.find(
+  (config) => config.chainId === CHAINID
+)
```

Comment:
So the attestation will always live in optimism network (that's why we have chainId), so no need to change it based on the active chain

Reply:
Then we have to deploy schema on all the chain + eas doesn't support mode :(

Reply:
For doing it we need to also make sure that the user needs to change the chain that they are connected to. Might need some more things to take care of => could we put this as a stretch goal and merge this version, I can have a look into this part after creating a simple profile page.

Reply:
> Then we have to deploy schema on all the chains

No, that's exactly what I'm saying, we don't want to deploy it on all chains, it will only be on a single chain, so no need for an array of `EAS_CHAIN_CONFIGS`, there will only be a single chain

So it's a nit